lib/src/stack/coap_blockwise_layer.dart

```diff
@@ -250,7 +253,12 @@ class CoapBlockwiseLayer extends CoapAbstractLayer {
         status.currentNUM = nextNum;
         status.currentSZX = block1.szx;
         final nextBlock = _getNextRequestBlock(exchange.request!, status);
-        nextBlock.token ??= response.token; // reuse same token
+        if (multicast) {
+          nextBlock.destination = response.source;
+          nextBlock.uriHost = response.source?.address.host;
+        } else {
+          nextBlock.token ??= response.token; // reuse same token
+        }
```

Comment:
Hmm, I just noticed that this code is probably not working properly yet (I focussed on block2 transfers for now). Maybe it would be better to remove for now until and add a working solution in a follow-up PR.

Reply:
I think I now found a better solution by refactoring and reusing the conversion from multicast to unicast :) I haven't tested it with block1 transfers yet, though. Let me know if this solution works for now – in the case of unicast requests everything should still work as before.